snap/snapcraft.yaml

```diff
@@ -29,8 +29,39 @@ apps:
       - network-bind
 
 parts:
+
+  lensfun:
+    plugin: cmake
+    source: https://github.com/lensfun/lensfun.git
+    source-commit: v0.3.2
+    configflags:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_PREFIX=/usr
+      - -DBINARY_PACKAGE_BUILD=On
+
+    build-packages:
+      - python3
+      - libglib2.0-dev
+      - libpng-dev
+      - doxygen
+      - python3-docutils
+
+    override-build: |
+      snapcraftctl build
+      # run lensfun-update-data and update lens data
+      PYTHONPATH=/usr/lib/python3.6/site-packages/ lensfun-update-data
```

Comment:
How long does this generally take? Thinking about maybe running it every time Darktable starts up, but not sure what kind of performance hit that might be.

Reply:
It would depend on your internet connection - I'd be tempted to avoid doing in on every start unless it fails fast.

I'm pretty new to snaps but there is some talk in the documentation of "space" to run "daemons" (as virtual root) - presumably there are some examples of using that "space" to run something analogous to cron (or indeed maybe just cron?).

Reply:
Yeah I agree, a daemon seems sensible. How often is the database updated, would you say? Daily? Weekly?